### PR TITLE
Fix a bug where decorator's `serviceAdded` is not invoked

### PR DESCRIPTION
**Pull request by ikhoon**
Motivation:

Checking the cause of CI failure, I found that `serviceAdded` is not called for the decorators directly added to a server. https://ge.armeria.dev/s/lqh63hxstqzou/tests/task/:xds:shadedTest/details/com.linecorp.armeria.xds.ConfigSourceGrpcServiceTest/testHeader()?top-execution=1

Previously, `serviceAdded` was only invoked for services and decorators wrapping the services. Decorators that were added dynamically were not being called:
- Route decorators
- Annotated decorators for Thrift
- Created by `DecoratingHttpServiceFunction`

Modifications:

- All internal decorators of `RouteDecoratingService` are correctly invoked.
  - A condition is added logic to prevent duplicate calls and infinitive calls.
- Added a similar approach to `THttpService`.
- Added `serviceAdded` method to `DecoratingHttpServiceFunction` to listen to service-added events
- Fixed an infinity loop where the unwrapped service is itself in `DecoratingService`

Result:

`Service.serviceAdded()` is now correctly invoked for dynamic decorators.

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/server/Server.java [hunk 1]
```diff
@@ -144,7 +144,9 @@ public static ServerBuilder builder() {
 
         // Invoke the serviceAdded() method in Service so that it can keep the reference to this Server or
         // add a listener to it.
-        config.serviceConfigs().forEach(cfg -> ServiceCallbackInvoker.invokeServiceAdded(cfg, cfg.service()));
+        for (ServiceConfig cfg : config.serviceConfigs()) {
+            ServiceCallbackInvoker.invokeServiceAdded(cfg, cfg.service());
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/server/Server.java:148]: It's just fixed for easier debugging. A line breaking is easier than lambda.

**Review by minwoox — APPROVED** [review]
👍 👍 👍

**Review by jrhee17 — APPROVED** [review]
👍 👍 👍

**Comment by minwoox** [comment]
👍 👍 👍

**Pull request merged**